net/ping/ping_test.go

```diff
@@ -58,7 +62,62 @@ func TestPinger(t *testing.T) {
 	})
 
 	const fakeDuration = 100 * time.Millisecond
-	p.handleResponse(fakeResponse, localhost, clock.Now().Add(fakeDuration))
+	p.handleResponse(fakeResponse, clock.Now().Add(fakeDuration), v4Type)
+
+	select {
+	case dur := <-r:
+		want := fakeDuration
+		if dur != want {
+			t.Errorf("wanted ping response time = %d; got %d", want, dur)
+		}
+	case <-ctx.Done():
+		t.Fatal("did not get response by timeout")
+	}
+}
+
+func TestV6Pinger(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skipf("IPv6")
+	}
```

Comment:
Is this the correct thing to do?

Reply:
It's an ok match for our CI, but I think we could do a quick probe instead such as:

```
if c, err := net.ListenPacket("udp6", "::1"); err != nil {
  t.Skipf("IPv6 not supported: %s", err)
} else {
  c.Close()
}
```